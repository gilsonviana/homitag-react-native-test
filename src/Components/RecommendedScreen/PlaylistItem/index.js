import * as React from 'react'
import { TouchableOpacity, View, Image, Text, StyleSheet } from 'react-native'

import constants from '../../../Constants'

const PlaylistItem = ({
    albumUri,
    numOfTracks
}) => {
    return (
        <TouchableOpacity style={styles.container}>
            <Image style={styles.image} resizeMethod="auto" resizeMode="contain" source={{uri: albumUri}} />
            <Text style={styles.track}>{numOfTracks} {numOfTracks > 1 ? 'Tracks' : 'Track'}</Text>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        height: 212
    },
    image: {
        flex: 1,
        width: undefined,
    },
    track: {
        marginVertical: 6,
        color: constants.colorGray,
        textAlign: 'center'
    }
})

export default PlaylistItem