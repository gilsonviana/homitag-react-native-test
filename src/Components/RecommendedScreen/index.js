import * as React from 'react'
import { View, FlatList, Text, Animated, StyleSheet } from 'react-native'
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux'

import constants from '../../Constants'

import PlaylistItem from './PlaylistItem'
import { getToken } from '../../Store/Auth/actions'
import { getRecommendations } from '../../Store/Recommendations/actions'

const RecommendedScreen = ({
    getToken,
    getRecommendations,
    recommendations
}) => {
    const { useState, useEffect } = React 
    
    const [scrollY] = useState(new Animated.Value(0))

    const HEADER_MAX_HEIGHT = 150;
    const HEADER_MIN_HEIGHT = 0;

    useEffect(() => {
        const tokenRequest = async () => {
            try {
                const token = await getToken()

                if (token) {
                    await getRecommendations(token)
                }
            } catch (e) {
                console.log("useEffect", e.response);   
            }
        }
        tokenRequest()
    }, [])

    return (
        <View style={styles.container}>
            <Animated.View>
                <LinearGradient colors={[constants.colorPrimary, constants.colorBlack]}>
                    <Text style={styles.title}>Recommendations</Text>
                    <Text style={styles.heading}>{recommendations.message}</Text>
                </LinearGradient>
            </Animated.View>
            <FlatList 
                scrollEventThrottle={16}
                onScroll={Animated.event(
                    [{
                        nativeEvent: {contentOffset: {y: scrollY}}
                    }], {
                        useNativeDriver: false
                    }
                )}
                numColumns={2}
                style={styles.list}
                data={recommendations.playlists}
                renderItem={({ item }) => <PlaylistItem numOfTracks={item.tracks.total} albumUri={item.images[0].url}/>}
                keyExtractor={(item) => item.id}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: constants.colorBlack,
    },
    title: {
        paddingTop: 70,
        marginBottom: 12,
        color: constants.colorGray,
        fontSize: 18,
        textAlign: 'center'
    },
    list: {
        flex: 1,
        marginHorizontal: 6
    },
    heading: {
        textAlign: "center",
        marginBottom: 22,
        color: constants.colorGray,
    }
})

const mapStateToProps = (state) => ({
    recommendations: state.recommendations
})

export default connect(mapStateToProps, { getRecommendations, getToken })(RecommendedScreen)