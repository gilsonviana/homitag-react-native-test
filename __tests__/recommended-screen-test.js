import 'react-native'
import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import RecommendedScreen from 'Components/RecommendedScreen'

const mockStore = configureStore([])

jest.useFakeTimers();

describe('Recommended screen', () => {
    let store;
    let Component;

    beforeEach(() => {
        store = mockStore({
            recommendations: {
                message: '',
                playlists: []
            }
        })

        Component = renderer.create(
            <Provider store={store}>
                <RecommendedScreen />
            </Provider>
        )
    })

    it('should render with initial state', () => {
        expect(Component.toJSON()).toMatchSnapshot()
    })
})