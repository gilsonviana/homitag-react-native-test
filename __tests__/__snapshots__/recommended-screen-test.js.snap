// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommended screen should render with initial state 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294091621,
      4281354078,
      4281354078,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "backgroundColor": "#222831",
      "flex": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ececec"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
      }
    }
  />
</BVLinearGradient>
`;
